### PR TITLE
Pull Request for Issue1464: Arbitrary motions of the hexapod

### PR DESCRIPTION
**Pull request by iainworkman**
Added:
- AM3DCoordinatedSystemControl ~ An abstract class which allows for coordination of three motors in some 3d space.
- AM3DRotatedSystemControl ~ Extends coordinated system to allow rotations of the coordinate system
- SGMHexapodTransformedAxis ~ Extends AM3DRotatedSystemControl to interact with the trajectory moves of the hexapod as if they were simple single axes motions.
- Added the ability to rotate the coordinate system of the SGMHexapod as a whole, and initialized it to hardcoded values which Teak determined was the plane of the sample plate.

Tweaked:
- Made changes at both the AM level of the hexapod and the driver level in order to make fine motions more accurate.

General AM Changes:
- Added a new match type (MatchWithinTolerance) to the AMControlWaitAction, and added the ability to (optionally) specify the match type when calling AMAcitonSupport::buildControlWaitAction().

### CODE REVIEW

**Comment by dretrex** [comment]
Are you wanting this to be reviewed now @iainworkman or would you like to hold off until testing has been completed?

**Comment by iainworkman** [comment]
I was just going to skim it myself for comments and debugs etc. Then I'll add the estimate and mark it for review.

**Comment by iainworkman** [comment]
I've tested it with a number of pretty find grain 2D scans, and it went through all 3 hours of them, so it seems okay. Will see how it behaves for Teak and Tom over the next few weeks.

File: source/application/AMPIC887DriverServer/AMPIC887EpicsCoordinator.cpp [hunk 1]
```diff
@@ -626,7 +331,9 @@ void AMPIC887EpicsCoordinator::onAllConnected(bool connectedState)
 
 void AMPIC887EpicsCoordinator::onTrajectoryMove()
 {
+	qDebug() << "Moving to :";
```
Comment (dretrex) [inline source/application/AMPIC887DriverServer/AMPIC887EpicsCoordinator.cpp]: Debugs you want to stay?

File: source/beamline/SGM/SGMHexapod.h [hunk 2]
```diff
@@ -3,11 +3,15 @@
 
 #include <QObject>
 #include "beamline/AMControlSet.h"
+
+#define HEXAPOD_POSITION_TOLERANCE 0.008
```
Comment (dretrex) [inline source/beamline/SGM/SGMHexapod.h]: Is the position tolerance 0.008 or 0.0008?  Based on our conversations I was under the impression it is the latter.

**Comment by dretrex** [comment]
This looks good to me.
Time Estimate: 20 minutes

File: source/application/AMPIC887DriverServer/AMPIC887EpicsCoordinator.cpp (hunk 1 above)
Comment (iainworkman) [inline source/application/AMPIC887DriverServer/AMPIC887EpicsCoordinator.cpp]: Yeah, makes it way easier to debug issues with the driver.

File: source/beamline/SGM/SGMHexapod.h (hunk 2 above)
Comment (iainworkman) [inline source/beamline/SGM/SGMHexapod.h]: At the driver level it's 0.0008. In AM I set it one order of magnitude greater.

Comment (dretrex) [inline source/beamline/SGM/SGMHexapod.h]: Ah yes, that's my bad.

**Pull request merged**